src/Common/DocumentInfo.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\CoreBundle\Common;
+
+use EMS\CommonBundle\Common\EMSLink;
+use EMS\CoreBundle\Entity\Revision;
+
+final class DocumentInfo
+{
+    private EMSLink $id;
+    /** @var Revision[] */
+    private array $revisions;
+
+    /**
+     * @param Revision[] $revisions
+     */
+    public function __construct(EMSLink $id, array $revisions)
+    {
+        $this->id = $id;
+        $this->revisions = $revisions;
+    }
+
+    public function getRevision(string $environmentName): ?Revision
+    {
+        foreach ($this->revisions as $revision) {
+            foreach ($revision->getEnvironments() as $environment) {
+                if ($environmentName === $environment->getName()) {
+                    return $revision;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    public function isAligned(string $environmentName): bool
```

Comment:
In a is Aligned I was expecting a Hash check ? endTime null only means the currentRevision